### PR TITLE
Issue #178: Take CSV of categories to hide from Remote Config

### PR DESCRIPTION
**Pull request by mcomella**
Yay, design can modify the categories themselves!

After this, #178 will become a UX-only bug (assuming they don't need anymore eng intervention).

### CODE REVIEW

File: Prox/Prox/Utilities/CategoriesUtil.swift [hunk 1]
```diff
@@ -6,51 +6,81 @@ import Foundation
...
+        let allMatched = categorySet.subtracting(HiddenCategories).isEmpty
+        return !allMatched
+    }
+
+    // UX gives us a CSV list of categories to hide and we hide them *and their children*.
+    // This is an exhaustive collection of these categories and their children.
+    // Caveat: we don't expect UX to include root categories, which are special-cased: we can reduce
+    // yelp resource use by whitelisting the categories we ask for meaning the other root categories
+    // are hidden implicitly.
+    static let HiddenCategories: Set<String> = {
```
Comment (fluffyemily) [inline Prox/Prox/Utilities/CategoriesUtil.swift]: nit: `hiddenCategories`. I know I've been bad at enforcing this but it's a new Swift 3.0 thing that constants are lowerCamelCase and it's taken me a while to get used to it.

File: Prox/Prox/Utilities/CategoriesUtil.swift [hunk 2]
```diff
@@ -13,6 +13,36 @@ struct CategoriesUtil {
...
+        let csv = RemoteConfigKeys.getString(forKey: RemoteConfigKeys.placeCategoriesToHideCSV)
+        return getHiddenCategories(forCSV: csv)
+    }()
+
+    // Separated for testing: I don't know how to do automated tests with the Firebase value.
+    internal static func getHiddenCategories(forCSV csv: String) -> Set<String> {
+        // We could be more robust
+        let categories = csv.characters.split(separator: ",")
+
```
Comment (jhugman) [inline Prox/Prox/Utilities/CategoriesUtil.swift]: In #241 I have added a method get string arrays out of RemoteConfig. This helps with ensuring accessing and splitting the string, and insist on setting a default.

File: Prox/Prox/RemoteConfigDefaults.plist [hunk 3]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>place_categories_to_hide_csv</key>
+	<string></string>
```
Comment (mcomella) [inline Prox/Prox/RemoteConfigDefaults.plist]: For now, yes – we don't know for sure what categories we want to filter so we should start at nothing.

**Comment by mcomella** [comment]
Waiting for #241 to land before landing this.

File: Prox/Prox/RemoteConfigDefaults.plist (hunk 3 above)
Comment (mcomella) [inline Prox/Prox/RemoteConfigDefaults.plist]: > ...but on second thought, maybe I should also filter on the client so we can get accurate data without relying on the server being correct.

This is less important than other things I can do so filed follow-up #251.

File: Prox/Prox/Utilities/CategoriesUtil.swift (hunk 2 above)
Comment (mcomella) [inline Prox/Prox/Utilities/CategoriesUtil.swift]: Decided not to use it because it'd be more work to switch over the already working code.

**Comment by mcomella** [comment]
This is blocked on #261. I don't want to rebase and land on top because it'll make backing it all out more difficult.

**Pull request merged**